pkg/syncer/fullsync.go

```diff
@@ -55,8 +55,12 @@ func csiFullSync(ctx context.Context, metadataSyncer *metadataSyncInformer) {
 	// For migrated volumes, invoke GetVolumeID from migration service.
 	for _, pv := range k8sPVs {
 		var volumeHandle string
-		var err error
-		if migrationFeatureStateForFullSync && pv.Spec.VsphereVolume != nil {
+		if pv.Spec.VsphereVolume != nil && migrationFeatureStateForFullSync {
+			volumeMigrationService, err := migration.GetVolumeMigrationService(ctx, &metadataSyncer.volumeManager, metadataSyncer.configInfo.Cfg)
```

Comment:
specifically, if you are calling `migration.GetVolumeMigrationService` in the loop, I suggest moving retrieval before the loop.

Reply:
Moved this outside and guarded with FSS check